### PR TITLE
Pell Grant 0 EFC and Marginal Rate from 2009 to 2023

### PR DESCRIPTION
**Pull request by CalebPena**
Fixes: #2885 

### Summary
:bookmark::card_file_box::chart_with_upwards_trend:

This pull request adds historical data and references for the Pell Grant EFC formula parameters, including the `automatic_zero` threshold and the `marginal_rate` brackets for different filing statuses. This enhances the policy engine's ability to model the Pell Grant program and its effects on student aid.

> _To make the policy engine better_
> _We updated the Pell Grant parameter_
> _We added `marginal_rate`_
> _And the 0 EFC state_
> _And we bumped up the version to a letter_

### Walkthrough
*  Add historical values and references for Pell Grant automatic 0 EFC threshold and marginal tax rate brackets ([link](https://github.com/PolicyEngine/policyengine-us/pull/2887/files?diff=unified&w=0#diff-3b3bc2777ea4d43a22c8e858a7e5896b8db1fbb354f5279167d4d499025bf2faL3-R30), [link](https://github.com/PolicyEngine/policyengine-us/pull/2887/files?diff=unified&w=0#diff-ae0ccbee6185871bf95e7e5af0323b7319f5c6dd3da9b6363ee2735aa4a0a8e9L4-R89), [link](https://github.com/PolicyEngine/policyengine-us/pull/2887/files?diff=unified&w=0#diff-ae0ccbee6185871bf95e7e5af0323b7319f5c6dd3da9b6363ee2735aa4a0a8e9L34-R126))
* Bump version number to minor update and update changelog ([link](https://github.com/PolicyEngine/policyengine-us/pull/2887/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))

### CODE REVIEW

File: policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
   unit: currency-USD
   label: Max income for an automatic 0 EFC
   reference:
-    - title: THE EFC FORMULA, 2022–2023
-      href: https://fsapartners.ed.gov/sites/default/files/2021-08/2223EFCFormulaGuide.pdf#page=5
+    - title: THE EFC FORMULA, 2009–2010
+      href: https://fsapartners.ed.gov/sites/default/files/attachments/efcformulaguide/111408EFCFormulaGuide0910.pdf#page=5
+    - title: THE EFC FORMULA, 2011–2012
+      href: https://fsapartners.ed.gov/sites/default/files/attachments/efcformulaguide/101310EFCFormulaGuide1112.pdf#page=5
+    - title: THE EFC FORMULA, 2012–2013
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml:19]: How do we reconcile these?
>New for 2012-2013: The income threshold for an automatic zero EFC has been increased from
$31,000 to $32,000 for the 2012-2013 Award Year. 

>New for 2013-2014: The income threshold for an automatic zero EFC has increased from
$23,000 to $24,000 for the 2013-2014 Award Year

File: policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml [hunk 2]
```diff
@@ -1,9 +1,30 @@
 description: The Department of Education sets the Expected Family Contribution to zero if a family has income below this value.
 values:
-  2023-01-01: 27_000
+  2009-08-01: 30_000
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml:3]: We're inconsistent about this but given the number of references, could you adopt this style to pair each value with the reference(s)?

https://github.com/PolicyEngine/policyengine-us/blob/6bdf9bb89ac989473eb80795ed2557b6f1206885/policyengine_us/parameters/gov/usda/snap/income/deductions/excess_shelter_expense/homeless/available.yaml#L20-L25

Comment (CalebPena) [reply policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml:3]: Do you want me to do this for the marginal rate as well? In that one, each reference would be in multiple places.

Comment (CalebPena) [reply policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml:3]: Do you want me to do this for the marginal rate as well? In that one, each reference would be in multiple places.

Comment (CalebPena) [reply policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml:3]: Will I have to use `parameter(period).gov.ed.pell_grant.efc.automatic_zero.value`

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml]: yes please

File: policyengine_us/parameters/gov/ed/pell_grant/efc/head/marginal_rate.yaml [hunk 3]
```diff
@@ -1,27 +1,162 @@
...
   - threshold:
-      2023-01-01: 0
+      2009-08-01: 0
     rate:
       2023-01-01: 0.22
   - threshold:
-      2023-01-01: 17_500
+      2009-08-01:
+        value: 14_200
+        meta:
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/ed/pell_grant/efc/head/marginal_rate.yaml:10]:
```suggestion
        metadata:
```

File: policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml (hunk 2 above)
Comment (MaxGhenis) [reply policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml]: ah shoot i see what you mean. applying this on scale parameters will create significant reference duplication.

@nikhilwoodruff what would you suggest here? we have many years of policy with changing values each year that we want to tie to specific references. should we make a separate scale for each year so we can attach the reference once for the full scale?

Comment (nikhilwoodruff) [reply policyengine_us/parameters/gov/ed/pell_grant/efc/automatic_zero.yaml]: Sorry for my delay- I think it's OK to add the reference at the highest level possible (such that all nodes below are from the reference). Filed https://github.com/PolicyEngine/policyengine-core/issues/115 which would then smooth out the parameter reference data.

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**